ext/date/php_date.c

```diff
@@ -5282,38 +5306,32 @@ static bool date_period_is_magic_property(zend_string *name)
 static zval *date_period_read_property(zend_object *object, zend_string *name, int type, void **cache_slot, zval *rv)
 {
 	if (type != BP_VAR_IS && type != BP_VAR_R) {
-		if (date_period_is_magic_property(name)) {
-			zend_throw_error(NULL, "Retrieval of DatePeriod->%s for modification is unsupported", ZSTR_VAL(name));
+		if (date_period_is_internal_property(name)) {
+			zend_throw_error(NULL, "Cannot modify readonly property DatePeriod::$%s", ZSTR_VAL(name));
 			return &EG(uninitialized_zval);
 		}
 	}
 
-	object->handlers->get_properties(object); /* build properties hash table */
-
 	return zend_std_read_property(object, name, type, cache_slot, rv);
 }
 /* }}} */
 
-/* {{{ date_period_write_property */
 static zval *date_period_write_property(zend_object *object, zend_string *name, zval *value, void **cache_slot)
 {
-	if (date_period_is_magic_property(name)) {
-		zend_throw_error(NULL, "Writing to DatePeriod->%s is unsupported", ZSTR_VAL(name));
+	if (zend_string_equals_literal(name, "current")) {
+		zend_throw_error(NULL, "Cannot modify readonly property DatePeriod::$%s", ZSTR_VAL(name));
 		return value;
 	}
 
 	return zend_std_write_property(object, name, value, cache_slot);
 }
-/* }}} */
 
-/* {{{ date_period_get_property_ptr_ptr */
 static zval *date_period_get_property_ptr_ptr(zend_object *object, zend_string *name, int type, void **cache_slot)
 {
-	if (date_period_is_magic_property(name)) {
-		zend_throw_error(NULL, "Retrieval of DatePeriod->%s for modification is unsupported", ZSTR_VAL(name));
+	if (zend_string_equals_literal(name, "current")) {
+		zend_throw_error(NULL, "Cannot modify readonly property DatePeriod::$%s", ZSTR_VAL(name));
```

Comment:
This is more a nit than anything else but we know the property name so why not just do:
```suggestion
		zend_throw_error(NULL, "Cannot modify readonly property DatePeriod::$current");
```
?
(I think this might require the use of a different ``zend_throw_`` function).

Reply:
> This is more a nit than anything else but we know the property name so why not just do:

IMO it's better not to hardcode the property name so if there's any change of behavior (i.e. a new property is added to the condition), the error message won't have to be modified.

> (I think this might require the use of a different zend_throw_ function).

What do you mean?

Reply:
FWIW, I agree that it is better not to hard code this.